packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -395,6 +435,15 @@ export async function activate(context: vscode.ExtensionContext) {
     sfdxProjectOpened
   );
 
+  // Set internal dev context
+  const internalDev = sfdxCoreSettings.getInternalDev();
+
+  vscode.commands.executeCommand(
+    'setContext',
+    'sfdx:internal_dev',
+    internalDev
+  );
+
   let defaultUsernameorAlias: string | undefined;
```

Comment:
Need to add more logic to activation to avoid enabling features that won't work in this dev mode e.g. org picker.